x-pack/legacy/plugins/fleet/server/mappings.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const mappings = {
+  agents: {
+    properties: {
+      shared_id: {
+        type: 'text',
+      },
+      type: {
+        type: 'text',
+      },
+      active: {
+        type: 'boolean',
+      },
+      enrolled_at: {
+        type: 'date',
+      },
+      access_token: {
+        type: 'text',
+      },
+      version: {
+        type: 'text',
+      },
+      user_provided_metadata: {
```

Comment:
This field should be a string, and we will store JSON to it. Like this will cause bugs when for one agent the metadata is { location: 'New York' } and another { location: 2 }. In this example, inserts / index / updates would fail. Dynamic is only dynamic the first time you index data

Reply:
Fixed Good catch